public/modules/users/controllers/settings.client.controller.js

```diff
@@ -18,7 +53,14 @@ angular.module('users').controller('SettingsController', ['$scope', '$http', '$l
 
 		// Check if provider is already in use with current user
 		$scope.isConnectedSocialAccount = function(provider) {
-			return $scope.user.provider === provider || ($scope.user.additionalProvidersData && $scope.user.additionalProvidersData[provider]);
+			var connected = false;
+			angular.forEach($scope.user.identities, function(identity) {
+				if(identity.provider === provider ) {
+					connected = true;
+					$scope.linkedProviders.push(provider);
```

Comment:
Why this $scope.linkedProviders.push(provider) line here. Seems like a sideeffect when the method only should check if user already has an identity with that provider.
Maybe populating this array on loading (like the $scope.identitiesToEmails() call) is a better approach and here only return the result of a $scope.linkedProviders.contains(provider).